components/binder/BinderDetails.tsx

```diff
@@ -0,0 +1,209 @@
+        <Box display="flex" justifyContent="space-between" mb={4}>
+          <Button colorScheme="blue" onClick={onUpdateOpen}>
+            Update Binder
+          </Button>
+          <Button colorScheme="red" onClick={onDeleteOpen}>
+            Delete Binder
+          </Button>
+        </Box>
+      )}
+
+      <SimpleGrid columns={{ base: 2, md: 3, lg: 5 }} spacing={4}>
+        {binderLoading
+          ? Array.from({ length: BINDER_CONSTANTS.ROWS_PER_PAGE }).map((_, index) => (
+              <Box key={index} textAlign="center">
+                <Skeleton height="200px" width="full" />
+                <Skeleton height="20px" mt="2" />
+                <Skeleton height="20px" mt="1" />
+              </Box>
+            ))
+          : currentCards.map((card, index) => (
+              <Box key={index} textAlign="center">
+                {card ? (
+                  <>
+                    <Image
+                      className="cursor-pointer select-none w-full max-w-xs sm:max-w-sm"
+                      src={`https://simulationhockey.com/tradingcards/${card.image_url}`}
+                      alt={card.player_name}
+                    />
+                    <Text>{card.player_name}</Text>
+                    <Text>{card.card_rarity}</Text>
+                  </>
+                ) : (
+                  <Image
+                    className="cursor-pointer select-none w-full max-w-xs sm:max-w-sm"
+                    src="/cardback.png"
+                    alt="Card Placeholder"
+                  />
+                )}
+              </Box>
+            ))}
+      </SimpleGrid>
+
+      {!binderLoading && (
+        <TablePagination
+          totalRows={totalRows}
+          rowsPerPage={BINDER_CONSTANTS.ROWS_PER_PAGE}
+          onPageChange={setCurrentPage}
+        />
+      )}
+
+      {/* Update Binder Modal */}
+      <Modal isOpen={isUpdateOpen} onClose={onUpdateClose} size="full">
+        <ModalOverlay />
+        <ModalContent maxW="90vw" maxH="90vh">
+          <ModalHeader className="bg-primary text-secondary">
+            Update Binder
+          </ModalHeader>
+          <ModalBody className="bg-primary text-secondary" overflow="auto">
+            <UpdateBinder
+              bid={bid}
+              currentCards={fullBinderData}
+              onClose={onUpdateClose}
+            />
+          </ModalBody>
+          <ModalFooter className="bg-primary text-secondary">
+            <Button colorScheme="blue" onClick={onUpdateClose}>
+              Close
+            </Button>
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
+
+      {/* Delete Confirmation Modal */}
+      <Modal isOpen={isDeleteOpen} onClose={onDeleteClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader className="bg-primary text-secondary">
+            Confirm Delete
+          </ModalHeader>
+          <ModalBody className="bg-primary text-secondary">
```

Comment:
The checkbox probably isn't necessary but I honestly don't care so feel free to leave it